tests/sentry/workflow_engine/endpoints/test_organization_workflow_group_history.py

```diff
@@ -0,0 +1,251 @@
+            )
+        self.group_2 = self.create_group()
+        self.history.append(
+            WorkflowFireHistory(workflow=self.workflow, group=self.group_2, event_id=uuid4().hex)
+        )
+        self.group_3 = self.create_group()
+        for i in range(2):
+            self.history.append(
+                WorkflowFireHistory(
+                    workflow=self.workflow,
+                    group=self.group_3,
+                    event_id=uuid4().hex,
+                )
+            )
+        self.workflow_2 = self.create_workflow(organization=self.organization)
+        self.history.append(
+            WorkflowFireHistory(workflow=self.workflow_2, group=self.group, event_id=uuid4().hex)
+        )
+
+        histories: list[WorkflowFireHistory] = WorkflowFireHistory.objects.bulk_create(self.history)
+
+        # manually update date_added
+        for i in range(3):
+            histories[i].update(date_added=before_now(days=i + 1))
+        histories[3].update(date_added=before_now(days=1))
+        for i in range(2):
+            histories[i + 4].update(date_added=before_now(days=i + 1))
+        histories[-1].update(date_added=before_now(days=0))
+
+        self.base_triggered_date = before_now(days=1)
+
+        self.login_as(self.user)
+
+    def assert_correct_pagination(self, rule, start, end, expected, cursor=None, per_page=25):
+        result = fetch_workflow_groups_paginated(rule, start, end, cursor, per_page)
+        assert result.results == expected, (result.results, expected)
+        return result
+
+    def test_workflow_groups_paginated(self):
+        self.assert_correct_pagination(
+            self.workflow,
+            before_now(days=6),
+            before_now(days=0),
+            [
+                WorkflowGroupHistory(
+                    self.group,
+                    count=3,
+                    last_triggered=self.base_triggered_date,
+                    event_id=self.history[0].event_id,
+                ),
+                WorkflowGroupHistory(
+                    self.group_3,
+                    count=2,
+                    last_triggered=self.base_triggered_date,
+                    event_id=self.history[4].event_id,
+                ),
+                WorkflowGroupHistory(
+                    self.group_2,
+                    count=1,
+                    last_triggered=self.base_triggered_date,
+                    event_id=self.history[3].event_id,
+                ),
+            ],
+        )
+        result = self.assert_correct_pagination(
+            self.workflow,
+            before_now(days=6),
+            before_now(days=0),
+            [
+                WorkflowGroupHistory(
+                    self.group,
+                    count=3,
+                    last_triggered=self.base_triggered_date,
+                    event_id=self.history[0].event_id,
+                ),
+            ],
+            per_page=1,
+        )
```

Comment:
this assertion isn't super clear what's being evaluated -- could we use kwargs or something to help illustrate the expected vs comparison results?